packages/database/src/migrations/20210607053528-AddPostcodeLevelDataInAU-modifies-data.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+import { generateId, codeToId, insertObject } from '../utilities';
+import POSTCODE_ARRAY from './migrationData/20210607053528-AddPostcodeLevelDataInAU.json';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const STATES = {
+  ACT: 'AU_Australian Capital Territory',
+  NT: 'AU_Northern Territory',
+  NSW: 'AU_New South Wales',
+  VIC: 'AU_Victoria',
+  QLD: 'AU_Queensland',
+  WA: 'AU_Western Australia',
+  TAS: 'AU_Tasmania',
+  SA: 'AU_South Australia',
+};
+
+const HIERARCHY_CODE = 'covidau';
+
+exports.up = async function (db) {
+  for (const data of POSTCODE_ARRAY) {
+    // Insert into entity table
+    const { postcode, state, long, lat } = data;
+    const entityCode = `AU_${state}_${postcode}`;
+    const isExisted = await codeToId(db, 'entity', entityCode);
+    // There are same postcodes for multiple area from JSON, but they have similar latitude and longitude, we can pick one of them.
+    if (isExisted) {
+      continue;
+    }
+    const entity = {
+      id: generateId(),
+      code: entityCode,
+      parent_id: await codeToId(db, 'entity', STATES[state]),
+      name: postcode,
+      type: 'postcode',
+      country_code: 'AU',
+      metadata: {
+        dhis: {
+          isDataRegional: true,
+        },
+        openStreetMaps: {},
+      },
+    };
+    const point = { type: 'Point', coordinates: [long, lat] };
+    await insertObject(db, 'entity', entity);
```

Comment:
We aren't gonna run into any issues with meditrak syncing or anything due to manually inserting entities via a migration like this are we? @nick-beyondessential you had to work on something similar recently with the covid fiji entity import right? If I recall you imported the entities via the admin panel correct?

Reply:
Yes best practise is to import via admin-panel for meditrak sideeffects, which may not apply in this particular case, mostly to do with adding facilies to tables etc, we can't be sure of catching all of them, so adding via admin is safer.

Reply:
Thanks @nick-beyondessential ! Could you share an example file or the right format for the import file with latitude and longitude?

Reply:
Something like this:

[importPostcodes.xlsx](https://github.com/beyondessential/tupaia/files/6613351/importPostcodes.xlsx)
https://beyond-essential.slab.com/posts/importing-entities-w8ewpm2b

Reply:
Seeing as you have already written the migration to input postcodes. Maybe one option would be to run locally then export entities and use that file as the input file.

Reply:
Thanks @nick-beyondessential !